draft-packages/badge/KaizenDraft/Badge/Badge.tsx

```diff
@@ -3,27 +3,40 @@ import classNames from "classnames"
 
 import styles from "./styles.module.scss"
 
-type Variant = "default" | "active" | "dark"
+type Variant = "default" | "active" | "dark" | "dot"
+type Size = "small" | "large"
 
 export interface BadgeProps {
-  readonly children: string
+  readonly children?: string
+  /**
+   * The "dark" variant is no longer in the UI kit
+   */
   readonly variant?: Variant
+  /**
+   * renders reversed colors. Use on purple background
+   */
   readonly reversed?: boolean
+  /**
+   * Supports "small" and "large" sizes - defaults to "small"
+   */
+  readonly size?: Size
 }
 
 export const Badge = (props: BadgeProps) => {
-  const { children, variant = "default", reversed } = props
+  const { children, variant = "default", reversed, size = "small" } = props
 
   return (
     <span
       className={classNames(styles.badge, {
         [styles.default]: variant === "default",
         [styles.active]: variant === "active",
         [styles.dark]: variant === "dark",
+        [styles.dot]: variant === "dot",
         [styles.reversed]: reversed,
+        [styles.large]: size === "large",
       })}
     >
-      {children}
+      {props.variant === "dot" ? "" : children}
```

Comment:
I'd add a quick unit test here asserting that a `dot` variant doesn't render children.
Btw, is this another one where TS could possibly help with "conditional types"? As in, `children` has to be `undefined` whenever `variant == "dot"`.